draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,26 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function name in the expression
+must be supported.
+Implementations MUST support the function extensions defined
+in this document, but are free to support or not support other registered function extensions.
```

Comment:
```suggestion
Implementations MUST support the function extensions defined
in this document, and SHOULD support other registered function extensions.
```

I think this gives a little more pressure to implement other registered functions but still an escape if they don't.

Reply:
I was trying to avoid being prescriptive, but I'd be comfortable with a statement like the above saying "[...] and SHOULD support all other registered function extensions".

Reply:
I've added the SHOULD requirement.

Reply:
I don't think we should say this, as there probably will be registrations for very platform- or application-specific functions.

Some IETF registries have a "recommended" column, which is kind of the inverse: if it is not recommended, you need a good reason to implement it at all.

I think at some point we'll roll up a set of registered functions and say that there is an expectation these will be available, kind of a JSONPath 1.1.  But remember that the largest pressure for extensions is in the projection/transformation space, and we don't even have an extension point for this.

Reply:
Ok, reverted the SHOULD change. At least I'm not now in a minority of one.

Reply:
> there probably will be registrations for very platform- or application-specific functions.

God, I hope not.  The registry should be a list of functions that should and _can_ be globally supported.  Platform- and application-specific function are by definition _not_ interoperable.